lib/beaker/dsl/install_utils/windows_utils.rb

```diff
@@ -0,0 +1,85 @@
+        #                  (Defaults the value of PUPPET_MASTER_SERVER)
+        #                  Requires Puppet 2.7.12 / PE 2.5.0
+        # @option msi_opts [String] PUPPET_AGENT_CERTNAME The node’s certificate name, and the name it uses when requesting catalogs. This will set a value for
+        #                  (Defaults to the node's fqdn as discovered by facter fqdn)
+        #                  Requires Puppet 2.7.12 / PE 2.5.0
+        # @option msi_opts [String] PUPPET_AGENT_ENVIRONMENT The node’s environment.
+        #                  (Defaults to production)
+        #                  Requires Puppet 3.3.1 / PE 3.1.0
+        # @option msi_opts [String] PUPPET_AGENT_STARTUP_MODE Whether the puppet agent service should run (or be allowed to run)
+        #                  (Defaults to Manual - valid values are Automatic, Manual or Disabled)
+        #                  Requires Puppet 3.4.0 / PE 3.2.0
+        # @option msi_opts [String] PUPPET_AGENT_ACCOUNT_USER Whether the puppet agent service should run (or be allowed to run)
+        #                  (Defaults to LocalSystem)
+        #                  Requires Puppet 3.4.0 / PE 3.2.0
+        # @option msi_opts [String] PUPPET_AGENT_ACCOUNT_PASSWORD The password to use for puppet agent’s user account
+        #                  (No default)
+        #                  Requires Puppet 3.4.0 / PE 3.2.0
+        # @option msi_opts [String] PUPPET_AGENT_ACCOUNT_DOMAIN The domain of puppet agent’s user account.
+        #                  (Defaults to .)
+        #                  Requires Puppet 3.4.0 / PE 3.2.0
+        # @option opts [Boolean] :debug output the MSI installation log when set to true
+        #                 otherwise do not output log (false; default behavior)
+        #
+        # @example
+        #  install_msi_on(hosts, 'c:\puppet.msi', {:debug => true})
+        #
+        # @api private
+        def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
+          block_on hosts do | host |
+            log_file = "#{Time.new.strftime('%Y-%m-%d_%H.%M.%S')}.log"
+            msi_opts['PUPPET_AGENT_STARTUP_MODE'] ||= 'Manual'
+            msi_params = msi_opts.map{|k, v| "#{k}=#{v}"}.join(' ')
+
+            # msiexec requires backslashes in file paths launched under cmd.exe start /w
+            url_pattern = /^(https?|file):\/\//
+            msi_path = msi_path.gsub(/\//, "\\") if msi_path !~ url_pattern
+
+            # msiexec requires quotes around paths with backslashes - c:\ or file://c:\
+            # not strictly needed for http:// but it simplifies this code
+            install_cmd = "start /w msiexec.exe /i \"#{msi_path}\" /qn /L*V #{log_file} #{msi_params}"
+            on host, Command.new(install_cmd, [], { :cmdexe => true })
+
+            if opts[:debug]
+              on host, Command.new("type #{log_file}", [], { :cmdexe => true })
+            end
+
+            if !host.is_cygwin?
```

Comment:
This was a carry-over from prior code, and it's the only thing I don't know if we need to keep or not?